### PR TITLE
Identifier tweaks

### PR DESCRIPTION
**Pull request by MikeInnes**
I’ve given all identifiers a scope + inspired by coffeescript highlighting, I’ve added highlighting for assignments.

Let me know if I should move this somewhere else; it seems to work but unfortunately I can’t run the tests.

### CODE REVIEW

**Comment by spencerlyon2** [comment]
You can't run them, or they fail?

**Comment by one-more-minute** [comment]
I can't run them at all – Atom's dev mode on my machine is totally broken so the specs window opens and crashes immediately.

**Comment by spencerlyon2** [comment]
Ha. Ok I'll pull and take a look this afternoon

**Comment by spencerlyon2** [comment]
So the tests are massively broken from this, but not because the grammar updates have any problems. We just have a lot more scope info and need to update the gets accordingly. 

One change is that all variable assignments are now highlighted. I'm not sold on whether or not I like this. If others would check out this branch and provide their opinion about the extra highlighting that's be great. Then we can decide which tests to fix in order to get this merged.

**Comment by one-more-minute** [comment]
Ok, I got rid of the assignments highlighting, that didn't work that well.

I've fixed up most of the tests, apart from a couple which seem to indicate problems. Namely, `B'` is not matched, `true` *is* matched (at least in an interpolation context) and keywords are marked as identifiers. The latter two are probably just ordering issues, would be good if you could give it a quick look.

As an aside, keyword highlighting seems pretty strange to me. At the moment `:x` is highlighted as a ternary operator followed by a keyword; shouldn't the whole thing just be a keyword?

File: grammars/julia.cson [hunk 1]
```diff
@@ -194,6 +194,10 @@ repository:
         name: "support.function.macro.julia"
       }
       {
+        match: "[[:alpha:]_][[:word:]!]*"
+        name: "support.identifier.julia"
```
Comment (spencerlyon2) [inline grammars/julia.cson]: This fails to match `B` in  `B'`.

I tried adjusting this to be:

```
      {
        match: "([[:alpha:]_][[:word:]!]*)'?"
        captures:
            "1":
                name: "support.identifier.julia"
      }
```

But it didn't help. Any ideas?

File: spec/julia-spec.coffee [hunk 2]
```diff
@@ -89,42 +90,42 @@ describe "Julia grammar", ->
...
+    expect(tokens[3]).toEqual value: ".", scopes: ["source.julia", "keyword.operator.dots.julia"]
+    expect(tokens[4]).toEqual value: "Test", scopes: ["source.julia", "support.identifier.julia"]
 
   it "tokenizes export statements", ->
     {tokens} = grammar.tokenizeLine("export my_awesome_function")
     expect(tokens[0]).toEqual value: "export", scopes: ["source.julia", "keyword.control.export.julia"]
-    expect(tokens[1]).toEqual value: " my_awesome_function", scopes: ["source.julia"]
+    expect(tokens[2]).toEqual value: "my_awesome_function", scopes: ["source.julia", "support.identifier.julia"]
 
   it "tokenizes symbols", ->
```
Comment (spencerlyon2) [inline spec/julia-spec.coffee]: I think we lost all symbol tokenization.

**Comment by spencerlyon2** [comment]
Not tokenizing symbols is what causes almost all the failures now. We might need to re-work how that is handled right now (you mentioned it is kinda weird)

**Comment by MikeInnes** [comment]
Let's leave this – it adds a fair bit of complication, and it turns out we can get decent token info just using regexes and Atom APIs.

**Pull request closed without merging**